base/cvd/cuttlefish/host/commands/cvd/selector/instance_group_record.cpp

```diff
@@ -59,20 +59,54 @@ Result<LocalInstanceGroup> LocalInstanceGroup::Create(
 
   for (const auto& instance_proto : group_proto.instances()) {
     instances.emplace_back(group_proto, instance_proto);
-    ids.insert(instance_proto.id());
+    auto id = instance_proto.id();
+    if (id != 0) {
+      // Only non-zero ids are checked, zero means no id has been assigned yet.
+      CF_EXPECTF(ids.find(id) == ids.end(),
+                 "Instances must have unique ids, foud {} repeated", id);
```

Comment:
nit: `found`

nit: use quotes or a separator around `{}` to distinguish the user-provided input from the error message.